mithril/mithril.d.ts

```diff
@@ -8,25 +8,30 @@
 interface MithrilStatic {
 	(selector: string, attributes: Object, children?: any): MithrilVirtualElement;
 	(selector: string, children?: any): MithrilVirtualElement;
-	prop(value?: any): (value?: any) => any;
+	prop<T>(value?: T): (value?: T) => T;
```

Comment:
m.prop can also return a property/promise hybrid thing, so I'd add the following:

    prop<T>(promise: MithrilPromise<T>) : MithrilPromiseProperty<T>;

    interface MithrilPromiseProperty<T> extends MithrilPromise<T> {
        (): T;
        (value: T): T;
        toJSON(): T;
    }